### ISSUE TITLE
Bump ring dependency to 0.17.0

### ISSUE THREAD

**Issue by grubles** [talaia-labs/rust-teos#243]
0.16.20 does not support ppc64le arch. The recently released 0.17.0 does. Is there anything that would prevent bumping the version?

**Comment by sr-gi** [comment]
`ring` is not a direct dependency for `teos`, but on the dependency chain of `rcgen` IIRC.

Looks like the latest version of `rcgen` (0.11.3) still uses ring 0.16. I wouldn't mind bumping `rcgen`'s version once it includes `ring` 0.17.0.

What is your use-case btw, deploying `teos` on a PowerPC?

**Comment by grubles** [comment]
Yes I have a beefy modern POWER machine I'd like to run a watchtower on for friends.

**Comment by grubles** [comment]
Looks like `rcgen` 0.12.0 is now released and includes `ring` 0.17. https://github.com/rustls/rcgen/blob/main/rcgen/CHANGELOG.md#release-0120---december-16-2023

**Comment by sr-gi** [comment]
@grubles would you mind giving https://github.com/talaia-labs/rust-teos/pull/248 a go?

**Comment by grubles** [comment]
If I build using `cargo install --path teos`, I still hit an error about `ring v0.16.20`. Should I be building a different way? Sorry, not super familiar with building rust projects. 

```
error: failed to run custom build command for `ring v0.16.20`

Caused by:
  process didn't exit successfully: `/home/user/builds/rust-teos/target/release/build/ring-893e2ab3b92e0036/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ring-0.16.20/build.rs:358:10:
  called `Option::unwrap()` on a `None` value
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: failed to compile `teos v0.2.0 (/home/user/builds/rust-teos/teos)`, intermediate artifacts can be found at `/home/user/builds/rust-teos/target`.
```

**Comment by sr-gi** [comment]
Ok, my bad. Looks like `rcgen` was not out only dependency that had a dependency chain with `ring 0.16`. Turns out `tonic` does too (a double one actually: `tonic` -> `rustls` - > `ring` and `tonic` -> `tokio-rustls` - > `rustls` -> `ring`).

The good news is that `rustls` v22 already uses `ring` v17 ([ref](https://github.com/rustls/rustls/blob/309a5d5051c42a15d2811a62b476335f36a66ae7/rustls/Cargo.toml#L21)), and the last version of `tokio-rustls` uses `rustls` v22 ([ref](https://github.com/rustls/tokio-rustls/blob/ff32e1ec589e3a81425d0721f9595e786231683d/Cargo.toml#L17)). The bad news is that tonic has not yet updated to any of them ([ref](https://github.com/hyperium/tonic/blob/48dd15e7a8db644d2c5be5d43d23839f288555c5/tonic/Cargo.toml#L83-L84)), so we are still stuck until our dependencies bumps theirs.

**Comment by grubles** [comment]
Ah bummer I was afraid that was the case. Okay I suppose we wait until tonic is updated then.

**Comment by sr-gi** [comment]
@grubles, all dependencies with `ring 0.16` should have been removed now in https://github.com/talaia-labs/rust-teos/pull/248. Would you mind giving that a go? CI failures should be unrelated

**Comment by grubles** [comment]
Great! Using 248 I am able to build now.

**Comment by sr-gi** [comment]
Can you try if the grpc client can properly connect to the server by issuing some `teos-cli` commands?

**Comment by sr-gi** [comment]
There was an issue with `teos-cli`, but it should be fixed now in the new version of #248
